src/runtime_src/hip/api/hipMemory.cpp

```diff
@@ -0,0 +1,222 @@
+    }
+    try 
+    {
+        xrt::core::hip::hipMalloc(ptr, size);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Allocates device accessible host memory.
+hipError_t 
+hipHostMalloc(void** ptr, size_t	size, unsigned int flags)
+{
+    *ptr = nullptr;
+    if (size == 0)
+    {
+        return hipSuccess;
+    }
+    try 
+    {
+        xrt::core::hip::hipHostMalloc(ptr, size, flags);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Free memory allocated by the hipHostMalloc().
+hipError_t
+hipHostFree(void* ptr)
+{
+    try 
+    {
+        xrt::core::hip::hipHostFree(ptr);
+        return hipSuccess;
+    }
+    catch 
+    (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Free memory allocated by the hipMalloc().
+hipError_t
+hipFree(void* ptr)
+{
+    try 
+    {
+        xrt::core::hip::hipFree(ptr);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Register host memory so it can be accessed from the current device.
+hipError_t
+hipHostRegister(void* hostPtr,
+    size_t sizeBytes,
+    unsigned int flags)
+{
+    try 
+    {
+        xrt::core::hip::hipHostRegister(hostPtr, sizeBytes, flags);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Un-register host pointer.
+hipError_t
+hipHostUnregister(void* hostPtr)
+{
+    try 
+    {
+        xrt::core::hip::hipHostUnregister(hostPtr);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Get Device pointer from Host Pointer allocated through hipHostMalloc.
+hipError_t
+hipHostGetDevicePointer(void** devPtr,
+                        void* hstPtr,
+                        unsigned int flags)
+{
+    try 
+    {
+        xrt::core::hip::hipHostGetDevicePointer(devPtr, hstPtr, flags);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Copy data from src to dst.
+hipError_t
+hipMemcpy(void* dst, 
+          const void* src,
+          size_t sizeBytes,
+          hipMemcpyKind kind)
+{
+    try 
+    {
+        xrt::core::hip::hipMemCpy(dst, src, sizeBytes, kind);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
```

Comment:
There is a lot of copy-paste of boiler-plate code all over the place. At the end this could be simplified to something like
```suggestion
{
    return handle_hip_error([&] { xrt::core::hip::hipMemCpy(dst, src, sizeBytes, kind); });
}
```

Reply:
> handle_hip_error

those functions are just stubs now and look identical, but they will have different error handling once implemented in later pull request.

Reply:
OK, it will be interesting to see how things go at that time then.